apps/cowswap-frontend/src/modules/tokensList/hooks/useSourceChainId.ts

```diff
@@ -1,14 +1,26 @@
+import { useMemo } from 'react'
+
 import { SupportedChainId } from '@cowprotocol/cow-sdk'
 import { useWalletInfo } from '@cowprotocol/wallet'
 
 import { useSelectTokenWidgetState } from './useSelectTokenWidgetState'
 
-export function useSourceChainId(): SupportedChainId {
+export function useSourceChainId(): { chainId: number; source: 'wallet' | 'selector' } {
   const { chainId } = useWalletInfo()
   const { selectedTargetChainId = chainId, open } = useSelectTokenWidgetState()
 
-  // Source chainId should always be a value from SupportedChainId
-  if (!open || !(selectedTargetChainId in SupportedChainId)) return chainId
+  return useMemo(() => {
+    // Source chainId should always be a value from SupportedChainId
```

Comment:
Comment outdated, no longer always a SupporteChainId

Reply:
Nope, it's still relevant, there is `!(selectedTargetChainId in SupportedChainId)` check